operator/pkg/controller/vault/vault_controller.go

```
@@ -227,6 +247,45 @@ func (r *ReconcileVault) Reconcile(request reconcile.Request) (reconcile.Result,
 		if err != nil {
 			return reconcile.Result{}, fmt.Errorf("failed to create secret for vault: %v", err)
 		}
+
+		// Distribute the CA certificate to every namespace defined
+		if len(v.Spec.CANamespaces) > 0 {
+
+			// Get the current version of the TLS Secret
+			var currentSecret corev1.Secret
+			err = r.client.Get(context.TODO(), client.ObjectKey{Name: sec.Name, Namespace: sec.Namespace}, &currentSecret)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to query current secret for vault: %v", err)
+			}
+
+			// We need the CA certificate only
+			delete(currentSecret.StringData, "server.crt")
+			delete(currentSecret.StringData, "server.key")
+			currentSecret.GetObjectMeta().SetUID("")
+
+			var namespaces []string
+
+			if v.Spec.CANamespaces[0] == "*" {
+				var namespaceList corev1.NamespaceList
+				if err := r.client.List(context.TODO(), &client.ListOptions{}, &namespaceList); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to list namespaces: %v", err)
+				}
+
+				for _, namespace := range namespaceList.Items {
+					namespaces = append(namespaces, namespace.Name)
+				}
+			} else {
+				namespaces = v.Spec.CANamespaces
```

Should we verify that the namespaces passed as list actually exist and only add add to the namespaces var those that do ? 

It would avoid failing shortly after if any of the specified namespaces do not exist

---

Maybe with a warning log that one of the namespaces was not found.

I feel that in the case of a wrong list of namespaces when a namespace does not exist failing hard is a bit too much since no workload would be expecting to find that secret anyway

---

Fixing this right now, thanks, good idea!